src/listeners/BoardListener.java

```diff
@@ -37,10 +56,13 @@ public void mouseClicked(MouseEvent e) {
                 }
                 firstClickRow = row;
                 firstClickCol = col;
+                // FOR TESTING PURPOSES ONLY (HIGHLIGHTING LEGAL MOVES)
+                boardPanel.setLegalMoveCoordinates(legalMoveCoordinates);
             } else {
                 boardPanel.eatPieces(firstClickRow, firstClickCol, row, col);
                 firstClickRow = -1;
                 firstClickCol = -1;
+                boardPanel.setLegalMoveCoordinates(new HashMap<>());
```

Comment:
Note that using this technique, We can't click a piece and show its legal moves then click another one. This means that the listener should remember the last piece clicked as well and act depending on the new clicked piece. However, we shouldn't worry about such details now as long as it works.